oms-cloudfoundry-solution/nested/omsCustomViews.json

```diff
@@ -87,16 +100,51 @@
         {
             "type": "Microsoft.Resources/deployments",
             "apiVersion": "2016-02-01",
-            "name": "[replace(replace(variables('omsviews').templates[copyIndex()], '%20', '_'), '_-_', '-')]",
+            "name": "[replace(replace(variables('omsviews').defailt[copyIndex()], '%20', '_'), '_-_', '-')]",
+            "copy": {
+                "name": "viewCopy",
+                "count": "[length(variables('omsviews').defailt)]",
+                "mode": "parallel"
+            },
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[concat(parameters('_artifactsLocation'), '/', variables('omsviews').defailt[copyIndex()], parameters('_artifactsLocationSasToken'))]",
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "resourcegroup": {
+                        "value": "[resourceGroup().name]"
+                    },
+                    "location": {
+                        "value": "[parameters('omsWorkspaceRegion')]"
+                    },
+                    "subscriptionId": {
+                        "value": "[subscription().subscriptionId]"
+                    },
+                    "workspace": {
+                        "value": "[parameters('omsWorkspaceName')]"
+                    },
+                    "workspaceapiversion": {
+                        "value": "2015-11-01-preview"
```

Comment:
this needs to be a literal value in the destination template, not passed in from the deployment - apiVersions can change the shape of the resource and a change here would likely break deployment.

Also, why are we not including all the dependent templates in the sample?

Reply:
They are actually OMS view file (.omsview) generated by OMS portal displaying and analyzing data forwarded from [OMS log analytics firehose nozzle](https://github.com/Azure/oms-log-analytics-firehose-nozzle), thus they are under the same repository as the nozzle.

This parameter is set required without a default value in these automatically generated view files(actually also templates). It seems it can only be passed in from the deployment.

Reply:
Ok, a couple things:
1) if we must pull the omsview files from another repo, then use a variable for the location with a context specific name (e.g. omsViewsLocation) so as not to confuse with the BPs in the repo.

Also, it looks like the files have a defaultValue of "" for the apiVersion, so you can remove them from here.  Passing in the apiVersion for a resource is an anti-pattern.

Reply:
Updated.

I tried to remove the parameter but seems they work like this by design. I should talk to OMS team about this.